src/IdentityInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authentication;
+
+use ArrayAccess;
+
+/**
+ * Identity interface
+ */
+interface IdentityInterface extends ArrayAccess
+{
+    /**
+     * Constructor
+     *
+     * @param array $identityData Identity data
+     * @param array $config Config options
+     */
+    public function __construct($identityData, array $config = []);
```

Comment:
Constructors in the interface can be a bit limiting. I wouldn't include this personally. If people want to vary the constructor they won't be able to.